Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -918,6 +934,24 @@ public void OnParseRequested(object requestor)
             }
         }
 
+        public bool OnSuspendParser(object requestor, Action busyAction, int millisecondsTimeout = 10000)
```

Comment:
This is a rather unintuitive default value. If I do not pass a timeout I would expect the request not to time out at all.

Anyway, setting a sensible timeout is really hard and very much depends on the project RD is dealing with.